### DISCUSSION TITLE
Manage store extensions fully with Composer

### DISCUSSION THREAD

**Discussion by shyim** [shopware/platform#3356] [RFC / Proposal]
## Effort

Low

## Description

Right now when a User installs a extension from the Store, we download the extension zip file and unpack it to `custom/plugins` or `custom/apps`. This brings a lot of problems with it:

- The download extension is not versioned with Composer
- We need to hack in Runtime the Classloader of Composer to load plugins namespaces, therefore can't use fully optimized class loader
- Plugin depending on another plugin cannot be automatically installed
- Mix of composer required plugins and without creates troubles when your plugin requires an another plugin and requires a additional composer package

## Benefits

- Switching also all store extensions to Composer, we have one straight way to install extensions. The UI behaves same as `composer require <package>`
- We can support installing depending plugins like Foundation plugins.
- We can use the [authoritative class maps](https://getcomposer.org/doc/articles/autoloader-optimization.md#optimization-level-2-a-authoritative-class-maps)

## Break strategy

When a extension updates which was before in `custom/plugins`  or `custom/apps`, we will delete that folder and require it through `store.shopware.com/xxx` with that correct version.

**Comment by OliverSkroblin** [comment]
Please provide an "advanced" error reporting to avoid support tickets because of miss configurations, not logged in, no auth.json, whatever ;)

**Comment by shyim** [comment]
We are going to reschedule the Composer Integration, as we are right now in the Migration to the Cloud with packages.shopware.com and want to improve the availability and performance of the system before we move all traffic to it. 

We plan to bring it in a minor version of 6.6, when the transition and optimisations have been completed.

**Reply by lasomethingsomething** [reply]
@shyim Do you still want to keep this discussion open or can we close?

**Reply by shyim** [reply]
As we anyway implement it, we can close it. we don't need any feedback anymore
